torchaudio/datasets/voxceleb1.py

```diff
@@ -0,0 +1,83 @@
+import os
+from pathlib import Path
+from typing import Tuple, Union
+
+import torchaudio
+from torch import Tensor
+from torch.hub import download_url_to_file
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import (
+    extract_archive,
+)
+
+
+_RELEASE_CONFIGS = {
+    "dev": {
+        "url": "https://thor.robots.ox.ac.uk/~vgg/data/voxceleb/vox1a/vox1_dev_wav.zip",
+        "checksum": "ff3b4ce606718a2d221299d21f1dded47097907762c6783e47fe823cad9f001e",
+    },
+    "test": {
+        "url": "https://thor.robots.ox.ac.uk/~vgg/data/voxceleb/vox1a/vox1_test_wav.zip",
+        "checksum": "8de57f347fe22b2c24526e9f444f689ecf5096fc2a92018cf420ff6b5b15eaea",
+    },
+}
+
+
+class VoxCeleb1(Dataset):
+    """Create *VoxCeleb1* [:footcite:`nagrani2017voxceleb`] Dataset.
+
+    Args:
+        root (str or Path): Path to the directory where the dataset is found or downloaded.
+        subset (str, optional): Subset of the dataset to use. Options: ["dev", "test"]. (Default: ``dev``)
+        download (bool, optional):
+            Whether to download the dataset if it is not found at root path. (Default: ``False``).
+    """
+
+    _ext_audio = ".wav"
+
+    def __init__(self, root: Union[str, Path], subset: str = "dev", download: bool = False) -> None:
+        assert subset in ["dev", "test"], "`subset` must be one of ['dev', 'test']"
+
+        # Get string representation of 'root' in case Path object is passed
+        root = os.fspath(root)
+
+        basename = os.path.basename(_RELEASE_CONFIGS[subset]["url"])
+        archive = os.path.join(root, basename)
+        self._path = os.path.join(root, basename.replace(".zip", ""))
+        if download:
```

Comment:
I noticed that roughly half our dataset implementations will check and throw a Runtime error if the dataset does not exist and download is False, while the other half will not do anything if download is set to False, which is the default value. I think it makes sense to throw an error (or at least warning) in those cases otherwise the dataset intialization does not make sense, thoughts?

cc @mthrok

Reply:
Yeah, throwing runtime error makes sense and seems more intuitive UX.

Reply:
Cool, thanks. We can add such UX to other datasets in a separate pr.